app/models/site.rb

```diff
@@ -49,8 +50,35 @@ class Site < ApplicationRecord
                    inclusion: { in: POSSIBLE_CODES, message: "must be A-Z, 0-9 or -" },
                    presence: true
 
+  geocoded_by :full_address
+
   after_commit -> { GeocodeJob.perform_later("Site", id) }, if: :needs_geolocation?
 
+  def needs_geolocation?
+    full_address.present? && (
+    latitude.nil? || longitude.nil? || address_changed?
+    )
+  end
+
+  def full_address
+    address = [address1, address2, address3, address4, postcode]
+
+    unless location_name == MAIN_SITE
```

Comment:
![image](https://user-images.githubusercontent.com/470137/71082985-f271c580-2189-11ea-9403-9831735b1c64.png)

Reply:
Yep fixed this now